src/utils/src/vm_memory.rs

```diff
@@ -142,6 +142,212 @@ pub fn mark_dirty_mem(mem: &GuestMemoryMmap, addr: GuestAddress, len: usize) {
     });
 }
 
+pub trait ReadVolatile {
```

Comment:
I am not sure with Rust best practices... would it make sense to take those in separate files? That may be easier to later upstream?

Reply:
Mh, yeah these were... smaller when I started working on it, that's why its all in the same file. Don't think its necessarily about of place here, its specific to vm-memory utils, so I'd just keep it here.